frontend/apps/filemanager/filemanagerfilesearcher.lua

```diff
@@ -73,9 +78,18 @@ function FileSearcher:setSearchResults()
     if keywords == "*" then -- one * to show all files
         self.results = self.files
     else
+        if not self.case_sensitive then
+            keywords = Utf8Proc.lowercase(keywords)
+        end
         for __,f in pairs(self.files) do
-            if string.find(string.lower(f.name), string.lower(keywords)) and string.sub(f.name,-4) ~= ".sdr" then
-                table.insert(self.results, f)
+            if self.case_sensitive then
+                if string.find(f.name, keywords) and string.sub(f.name,-4) ~= ".sdr" then
+                    table.insert(self.results, f)
+                end
+            else
+                if string.find(Utf8Proc.lowercase(f.name), keywords) and string.sub(f.name,-4) ~= ".sdr" then
```

Comment:
We compare our search pattern with string.find here. So the lua patterns https://www.lua.org/pil/20.2.html are used for search. This is maybe not what the user expects when using wildcards with filenames.
Should we change that to be nearer to the standard linux glob? Insert a `.` before `?` and `*` if it is not there?

Reply:
Some opinions on that https://github.com/koreader/koreader/pull/7362.
I'd tend to standard filename wildcards.